core/src/test/scala/unit/kafka/admin/RackAwareTest.scala

```diff
@@ -0,0 +1,79 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.admin
+
+import scala.collection.{Map, Seq, mutable}
+import org.junit.Assert._
+
+trait RackAwareTest {
+
+  def ensureRackAwareAndEvenDistribution(assignment: Map[Int, Seq[Int]],
+                                         brokerRackMapping: Map[Int, String],
+                                         numBrokers: Int,
+                                         numPartitions: Int,
+                                         replicationFactor: Int,
+                                         verifyRackAware: Boolean = true,
+                                         verifyLeaderDistribution: Boolean = true,
+                                         verifyReplicasDistribution: Boolean = true) {
```

Comment:
Should we also verify that no two replicas from the same partition are assigned to the same broker?